File: backend/src/main/java/com/example/aidemo/user/controller/UserController.java
package com.example.aidemo.user.controller;

import com.example.aidemo.auth.dto.UserInfoResponse;
import com.example.aidemo.auth.service.AuthService;
import com.example.aidemo.common.Result;
import com.example.aidemo.security.CurrentUser;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * 用户控制器
 */
@RestController
@RequestMapping("/api/v1/user")
@RequiredArgsConstructor
public class UserController {

    private final AuthService authService;

    @GetMapping("/info")
    public Result<UserInfoResponse> getUserInfo(@CurrentUser Long userId) {
        UserInfoResponse response = authService.getUserInfo(userId);
        return Result.success(response);
    }
}

File: node-bff/src/ai/controllers/user.controller.ts
import { Controller, Get } from '@nestjs/common';
import { ApiProxyService } from '../api-proxy/services/api-proxy.service';

/**
 * 用户控制器
 * 负责处理用户信息相关的接口
 */
@Controller('user')
export class UserController {
  constructor(private readonly apiProxyService: ApiProxyService) {}

  /**
   * 获取用户信息
   */
  @Get('info')
  async getUserInfo() {
    return this.apiProxyService.getUserInfo();
  }
}